src/olympia/applications/tests.py

```diff
@@ -29,6 +29,158 @@ def test_appversions_feed(self):
+        response = self.client.put(self.url)
+        assert response.status_code == 201
+        assert AppVersion.objects.count() == 2
+        assert AppVersion.objects.filter(
+            application=amo.FIREFOX.id, version='42.0').exists()
+        assert AppVersion.objects.filter(
+            application=amo.FIREFOX.id, version='42.*').exists()
+
+    def test_release_android(self):
+        self.url = reverse_ns(
+            'appversions',
+            kwargs={'application': 'android', 'version': '84.0'})
+        response = self.client.put(self.url)
+        assert response.status_code == 201
+        assert AppVersion.objects.count() == 2
+        assert AppVersion.objects.filter(
+            application=amo.ANDROID.id, version='84.0').exists()
+        assert AppVersion.objects.filter(
+            application=amo.ANDROID.id, version='84.*').exists()
+
+    def test_alpha(self):
+        self.url = reverse_ns(
+            'appversions',
+            kwargs={'application': 'firefox', 'version': '42.0a1'})
+        response = self.client.put(self.url)
+        assert response.status_code == 201
+        assert AppVersion.objects.count() == 3
+        assert AppVersion.objects.filter(
+            application=amo.FIREFOX.id, version='42.0').exists()
+        assert AppVersion.objects.filter(
+            application=amo.FIREFOX.id, version='42.0a1').exists()
+        assert AppVersion.objects.filter(
+            application=amo.FIREFOX.id, version='42.*').exists()
+
+    def test_alpha_already_exists(self):
+        AppVersion.objects.create(application=amo.FIREFOX.id, version='42.0a1')
+        self.url = reverse_ns(
+            'appversions',
+            kwargs={'application': 'firefox', 'version': '42.0a1'})
+        response = self.client.put(self.url)
+        assert response.status_code == 201
+        assert AppVersion.objects.count() == 3
+        assert AppVersion.objects.filter(
+            application=amo.FIREFOX.id, version='42.0').exists()
+        assert AppVersion.objects.filter(
+            application=amo.FIREFOX.id, version='42.0a1').exists()
+        assert AppVersion.objects.filter(
+            application=amo.FIREFOX.id, version='42.*').exists()
+
+    def test_alpha_release_already_exists(self):
+        AppVersion.objects.create(application=amo.FIREFOX.id, version='42.0')
+        self.url = reverse_ns(
+            'appversions',
+            kwargs={'application': 'firefox', 'version': '42.0a1'})
+        response = self.client.put(self.url)
+        assert response.status_code == 201
+        assert AppVersion.objects.count() == 3
+        assert AppVersion.objects.filter(
+            application=amo.FIREFOX.id, version='42.0').exists()
+        assert AppVersion.objects.filter(
+            application=amo.FIREFOX.id, version='42.0a1').exists()
+        assert AppVersion.objects.filter(
+            application=amo.FIREFOX.id, version='42.*').exists()
+
+    def test_alpha_star_already_exists(self):
```

Comment:
This name suggests that this is a test for when the star version already exists, but you don't seem to include the setup code that pre-creates the star version. 

This also suggests that there may not be sufficient assertions in these tests if it's critical to know that *only* the non-pre-created versions have been created, which is why this test currently passes. Having said that, I'm not sure it's terribly important to know that.

Reply:
You're right, I forgot to pre-create the star variant here. I think we do have enough assertions though: the main goal of those tests is to ensure that a) we don't accidentally create more app versions that necessary and b) we don't somehow raise an exception when one already exists. So pre-creating the missing `AppVersion` instance like I do in the tests above and below that one should be enough.